### DISCUSSION TITLE
MIMO Engine Airpath Control (Project 45)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#10] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project #45](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/tree/main/projects/MIMO%20Engine%20Airpath%20Control)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by YorkPatty** [comment]
Hello,

My team and I have worked on this project and recently submitted a repository containing our MATLAB and Simulink code for a drop-in MPC for engine airpath control (https://github.com/YorkPatty/T513---SIEngineDynamometer).  While the original goal of the project is to implement a MIMO control solution, our implemented controller can control throttle and wastegate individually, but not together to output a single matrix of improved throttle and wastegate positions. Although we did not achieve a MIMO system, we established an excellent framework to continue expanding on the MPC control of this system, and are confident our work can be used as stepping stones for the continuation of this research. 
Good luck all!

**Reply by mw-agrace** [reply]
Thanks Patrick! Excellent contribution and first **MathWorks Excellence in Innovation** project submission! This will be a great resource for others looking to build on your work!
